modules/core/TextureMarble.cpp

```diff
@@ -0,0 +1,62 @@
+//  Created by Mahmoud El Bergui on 21.04.22.
+//
+
+#include "TextureMarble.h"
+#include "PerlinNoise.h"
+
+namespace {
+
+     Vec3f marbleMap(float value){
+               
+          Vec3f white = Vec3f(1,1,1);
+          Vec3f blue = Vec3f(0.6 , 0.53 , 0.467);
+          Vec3f color = (Vec3f) ((white - blue) * value + blue);
+               
+               return color;
+     }
+     
+     double smoothstep(float val){
+          return val * val * val *(val * (val * 6 -15) + 10);
+     }
+     
+         Vec3f mix(Vec3f& C0, Vec3f& C1, float f ){
```

Comment:
You can use this method inside the `marbleMap` method to make it nicer.

Reply:
I did the modification